performance/include/traccc/performance/details/is_same_object.hpp

```diff
@@ -0,0 +1,51 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/definitions/common.hpp"
+#include "traccc/definitions/primitives.hpp"
+
+// System include(s).
+#include <functional>
+
+namespace traccc::details {
+
+/// Generic unary functor for checking equality with a certain object
```

Comment:
*unitary

Reply:
😛 While this naming practically died out of modern C\+\+ by now, this is how such functors used to be called.

https://en.cppreference.com/w/cpp/utility/functional/unary_function